src/System.Management.Automation/engine/parser/TypeInferenceVisitor.cs

```diff
@@ -1983,8 +1983,19 @@ private void InferTypeFrom(VariableExpressionAst variableExpressionAst, List<PST
                     {
                         if (switchErrorStatement.Conditions?.Count > 0)
                         {
-                            parent = switchErrorStatement.Conditions[0];
+                            if (switchErrorStatement.Conditions[0].Extent.EndOffset < variableExpressionAst.Extent.StartOffset)
```

Comment:
could it also _not_ be the first condition which causes an issue?

Reply:
> could it also _not_ be the first condition which causes an issue?

So `Condition` is actually *only* the part here `switch (thisPart)`, everything else goes to `Bodies`. I don't think it's possible for a switch error statement to contain more than one AST there, even `switch ($_.Proc; $_)` doesn't (see below).

![image](https://github.com/PowerShell/PowerShell/assets/24977523/e26a67c4-8361-421d-96b3-502c363f9db6)

But I suspect you're right that it could also be an issue after the `Condition`

Reply:
`Get-Process | Sort-Object -Property {switch ($_.Proc) { a { $_.Proc<Tab>` does not crash. It also doesn't provide any completions but that's expected because it can't infer the type of the condition because it refers to a nonexistent property. If you change it to: `Get-Process | Sort-Object -Property {switch ($_.ProcessName) { a { $_.<Tab>` it works as expected.
As for the condition, I agree with @SeeminglyScience it shouldn't be possible for there to be more than 1.